src/Configuration/Config.AzureKeyVault/src/AzureKeyVaultConfigurationProvider.cs

```diff
@@ -66,7 +66,7 @@ private async Task PollForSecretChangesAsync()
 
         protected virtual async Task WaitForReload()
         {
-            await Task.Delay(_reloadInterval.Value.Milliseconds, _cancellationToken.Token);
+            await Task.Delay(_reloadInterval.Value, _cancellationToken.Token);
```

Comment:
What if `_reloadInterval` is `null`?

Reply:
To be precise: In this class this usage is safe, as `WaitForReload` is only called from `PollForSecretChangesAsync` and this (only) call is guarded: https://github.com/aspnet/Extensions/blob/43d453af6cb5a1087b95ab0b1d5249303a742d75/src/Configuration/Config.AzureKeyVault/src/AzureKeyVaultConfigurationProvider.cs#L127-L129

But the access modifier for this method is `protected`, hence (in theory) a sub-class can call `WaitForReload`, which may result in a NRE.

Reply:
Agree. I've converted `_reloadInterval` private field from `Nullable<TimeSpan>` to a regular `TimeSpan` to avoid NRE altogether. Please review.

Reply:
The non-nullable fields looks good.

Minor concern / thought: 
When not given as ctor-argument, its value is `TimeSpan.Zero`, so actually no "wait for reload". 
Should this be handled or is it fine to leave as is (as corner-case)?